frontend/src/metabase/ErrorBoundary.tsx

```diff
@@ -29,24 +27,20 @@ export default class ErrorBoundary extends React.Component<
     // if we don't provide a specific onError action, the component will display a generic error message
     if (this.props.onError) {
       this.props.onError(errorInfo);
-    } else {
       this.setState({
-        hasError: true,
-        errorDetails: errorInfo.componentStack,
+        hasError: false,
```

Comment:
The problem with the code in `componentDidCatch` is that the way this component works before, makes `componentDidCatch` never get called in some cases like when our DB is blowing up for instance.

Here's an explanation of how it worked previously (which blows up the FE)

1. `getDerivedStateFromError` is called setting `state.hasError: true`
2. since we also pass `onError`, this component will render `children`
    https://github.com/metabase/metabase/blob/0c56664819bb84a7a317caf0fc393d2fa1bf3ccd/frontend/src/metabase/App.tsx#L99
3. And since the children cause an error, this component throws the same error up, but since this is the uppermost `<ErrorBoundary />` no other component will catch the error this one thrown causing the application to blow up. So, `componentDidCatch` is never called, because it's called [after the render phase](https://beta.reactjs.org/reference/react/Component#componentdidcatch) (It's not clear to me when `componentDidCatch` will be called from the doc, but seems like the error boundary should never throw any error to make `componentDidCatched` being called. 

The solution is to ensure the uppermost `<ErrorBoundary />` never throws any error.

This is how this component is modified.
1. Always, render a fallback error component first. 
2. This makes sure `componentDidCatch` will be called.
3. `onError` will set `errorPage` which will be rendered here.
    https://github.com/metabase/metabase/blob/0c56664819bb84a7a317caf0fc393d2fa1bf3ccd/frontend/src/metabase/App.tsx#L108
4. But we need to render the children so `errorPage` could be rendered (otherwise, we'll only see the error rendered from this component), that's why we now set `hasError: false`